### ISSUE TITLE
qwen2-72b-gptq-int4的问题

### ISSUE THREAD

**Issue by lxb0425** [QwenLM/Qwen2#774]
swift 使用A100 训练了qwen2-72b-instruct  之后导出量化了qwen2-72b-gptq-int4-chat1.0
使用4张4090部署报错
![image](https://github.com/user-attachments/assets/e010fc3c-c4c7-4237-a79c-89980075751d)
我注意到conf.json的值为
intermediate_size": 29568
原始的模型是intermediate_size": 29696
我应该使用swift怎么量化导出微调后的模型才是29696这个值

**Comment by jklj077** [comment]
duplicate of https://github.com/QwenLM/Qwen2/issues/578

**Comment by lxb0425** [comment]
我使用了这个脚本
Qwen/Qwen2-72B-Instruct替换成了量化后导出的gptq模型 最后转成一个41G的pytorch_model.bin文件
import torch
from torch.nn import functional as F

from transformers import AutoModelForCausalLM

# must use AutoModelForCausalLM
model = AutoModelForCausalLM.from_pretrained("Qwen/Qwen2-72B-Instruct", torch_dtype="auto")

# this size is Qwen2-72B only
pad_size = 128 

sd = model.state_dict()

for i, k in enumerate(sd):
    v = sd[k]
    print(k, i)
    # interleaving the padded zeros
    if ('mlp.up_proj.weight' in k) or ('mlp.gate_proj.weight' in k):
        prev_v = F.pad(v.unsqueeze(1), (0, 0, 0, 1, 0, 0)).reshape(29568*2, -1)[:pad_size*2]
        new_v = torch.cat([prev_v, v[pad_size:]], dim=0)
        sd[k] = new_v
    elif 'mlp.down_proj.weight' in k:
        prev_v= F.pad(v.unsqueeze(2), (0, 1)).reshape(8192, 29568*2)[:, :pad_size*2]
        new_v = torch.cat([prev_v, v[:, pad_size:]], dim=1)
        sd[k] = new_v

# this is a very large file; make sure your RAM is enough to load the model
torch.save(sd, '/path/to/padded_model/pytorch_model.bin')
还是报错 
请问下这个路径错了吗

**Comment by jklj077** [comment]
use that script to pad the model parameters first then quantize the model
